server/client.go

```diff
@@ -2635,10 +2640,15 @@ func (c *client) processPingTimer() {
 
 	c.Debugf("%s Ping Timer", c.typeString())
 
-	// If we have had activity within the PingInterval no
-	// need to send a ping.
-	if delta := time.Since(c.last); delta < c.srv.getOpts().PingInterval {
-		c.Debugf("Delaying PING due to activity %v ago", delta.Round(time.Second))
+	// If we have had activity within the PingInterval then
+	// there is no need to send a ping. This can be client data
+	// or if we received a ping from the other side.
+	pingInterval := c.srv.getOpts().PingInterval
+
+	if delta := time.Since(c.last); delta < pingInterval {
+		c.Debugf("Delaying PING due to client activity %v ago", delta.Round(time.Second))
+	} else if delta := time.Since(c.ping.last); delta < pingInterval {
```

Comment:
The problem I see here is that we use PING/PONG to compute/update the connection's RTT (I know I use that for gateways). With this change, I am not sure a connection RTT would be ever be computed (except in monitor/events).

Reply:
Hmm, that is a good point. Let me think about this one for a bit.

Reply:
ok take a look now.